File: java/sodaslurper.java
import java.util.Scanner;

public class sodasurpler {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int e = scan.nextInt();
		int f = scan.nextInt();
		int c = scan.nextInt();
		int bottles = e + f;
		int remainings = bottles - c;
		int soda = 0;
		if (remainings >= 0)
			soda = 1 + (remainings / (c - 1));
		System.out.print(soda);
		scan.close();
	}
}

File: ts/src/sodaslurper.ts
import readline = require('readline')

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    let [empty, found, drink] = line.split(' ').map(Number)
    empty ??= 0
    found ??= 0
    drink ??= 0

    const bottles = empty + found
    if (bottles < drink) {
      console.log(0)
    } else {
      const soda = Math.floor((bottles - 1) / (drink - 1))
      console.log(soda)
    }
  })
  .on('close', () => {
    process.exit(0)
  })
